### PR TITLE
make dom_struct derive HeapSizeOf

### LINKED ISSUE #7357: Make dom_struct derive HeapSizeOf
Currently all of our DOM types have `#[dom_struct]` and `#[derive(HeapSizeOf)]`. We should make `#[dom_struct]` add that annotation automatically instead.

Code: `expand_dom_struct` in components/plugins/jstraceable.rs

### PR DESCRIPTION
**Pull request by jxs**
closes #7357

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7361)

### CODE REVIEW

File: components/script/dom/treewalker.rs [hunk 1]
```diff
@@ -22,8 +22,10 @@ use std::rc::Rc;
 pub struct TreeWalker {
     reflector_: Reflector,
     root_node: JS<Node>,
+    #[ignore_heap_size_of = "Defined in std"]
```
Comment (Ms2ger) [inline components/script/dom/treewalker.rs]: Neither of those are defined in std. lgtm otherwise.

File: components/script/dom/bindings/js.rs [hunk 2]
```diff
@@ -207,6 +207,7 @@ impl MutHeapJSVal {
 /// `JS<T>`.
 #[must_root]
 #[derive(JSTraceable)]
+#[derive(HeapSizeOf)]
```
Comment (jxs) [inline components/script/dom/bindings/js.rs]: added derive for MutHeap

File: components/script/dom/treewalker.rs [hunk 3]
```diff
@@ -24,6 +24,7 @@ pub struct TreeWalker {
     root_node: JS<Node>,
     current_node: MutHeap<JS<Node>>,
     what_to_show: u32,
+    #[ignore_heap_size_of = "function pointers and Rc<T> are hard"]
```
Comment (jxs) [inline components/script/dom/treewalker.rs]: added ignore_heap_size for the Filter enum

**Comment by jxs** [comment]
thanks @Ms2ger, updated with anotations

**Comment by Ms2ger** [comment]
Thank you for your PR!

@bors-servo r+

**Pull request merged**